### PR TITLE
[TSE] Implement the Strategy Execution Worker

### LINKED ISSUE #15: [TSE] Implement the Strategy Execution Worker
The goal of this task is to implement a worker that will execute and process active trading strategies. This worker will be responsible for fetching all active strategies from the database and processing them at regular intervals. Each strategy should be executed asynchronously to ensure efficient and timely processing.

### To Do
- Fetch Active Strategies:
  - Implement logic to query the database and retrieve all active trading strategies.

- Asynchronous Execution:
  - Design the worker to execute each strategy asynchronously.
  - Ensure that multiple strategies can be processed concurrently without blocking or delay.

- Error Handling:
  - Add robust error handling to manage any issues that arise during strategy execution.
  - Log any errors or exceptions for debugging purposes.

### Deliverables
- [ ] All tasks completed
- [ ] Unit tests
- [ ] Documentation

### PR DESCRIPTION
**Pull request by MWBlocky**
## Description

This PR introduces a worker responsible for executing and processing active trading strategies. The worker fetches all active strategies from the database and processes them asynchronously at regular intervals.

## Summary of Changes

- Added a new `ExecutionWorkerService` that handles the fetching and processing of active trading strategies.
- The service uses a cron job scheduled to run every 30 seconds to initiate the strategy execution process.
- Integrated `ArbitrageService` and `MarketMakingService` to fetch running arbitrage and market-making strategies, respectively.
- Integrated `ArbitrageStrategy` and `MarketMakingStrategy` to start the execution of the fetched strategies asynchronously.
- Added logging to track when the worker starts processing strategies.

## Related Issues

Fixes #15

### CODE REVIEW

File: packages/tse-backend/src/modules/strategy-execution/execution-worker.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+          checkIntervalSeconds: 60,
+          numberOfLayers: 5,
+          priceSourceType: PriceSourceType.MID_PRICE,
+          amountChangePerLayer: 50.00,
+          amountChangeType: AmountChangeType.FIXED,
+          ceilingPrice: 50000.00,
+          floorPrice: 30000.00,
+          createdAt: new Date(),
+          updatedAt: new Date(),
+          status: StrategyInstanceStatus.CREATED,
```
Comment (piotrswierzy) [inline packages/tse-backend/src/modules/strategy-execution/execution-worker.service.spec.ts:101]: Maybe extract test data creation to a fixture of some sort

File: packages/tse-backend/src/modules/strategy-execution/execution-worker.service.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  private logger = new Logger(ExecutionWorkerService.name);
+
+  constructor(
+    private readonly arbitrageService: ArbitrageService,
+    private readonly arbitrageStrategy: ArbitrageStrategy,
+    private readonly marketMakingService: MarketMakingService,
+    private readonly marketMakingStrategy: MarketMakingStrategy
+  ) {}
+
+  @Cron(CronExpression.EVERY_30_SECONDS)
```
Comment (piotrswierzy) [inline packages/tse-backend/src/modules/strategy-execution/execution-worker.service.ts:19]: The expression should be taken from env

File: packages/tse-backend/src/modules/strategy-execution/execution-worker.service.spec.ts (hunk 1 above)
Comment (MWBlocky) [reply packages/tse-backend/src/modules/strategy-execution/execution-worker.service.spec.ts]: Fixed: 4f660478

File: packages/tse-backend/src/modules/strategy-execution/execution-worker.service.ts (hunk 2 above)
Comment (MWBlocky) [reply packages/tse-backend/src/modules/strategy-execution/execution-worker.service.ts]: Fixed: 4f660478

**Review by piotrswierzy — APPROVED** [review]

**Pull request merged**